### ISSUE TITLE
Dashboard doesn't work in Vivaldi after extension update

### ISSUE THREAD

**Issue by lainverse** [Tampermonkey/tampermonkey#1148]
## Expected Behavior
When Dashboard link from menu or Options from extension Details clicked Dashboard should open.
When Edit clicked from script's menu it have to open for edit.

## Actual Behavior
This site can’t be reached
The webpage at chrome-extension://gcalenpjmijncebpfijmoaglllgpjagf/options.html#url=aHR0cHM6Ly93d3cudGFtcGVybW9ua2V5Lm5ldC9jaGFuZ2Vsb2cucGhwP3ZlcnNpb249NC4xMi42MTI3JmV4dD1nY2Fs&nav=dashboard might be temporarily down or it may have moved permanently to a new web address.
ERR_INVALID_RESPONSE

This site can’t be reached
The webpage at chrome-extension://gcalenpjmijncebpfijmoaglllgpjagf/options.html#url=&nav=dashboard might be temporarily down or it may have moved permanently to a new web address.
ERR_INVALID_RESPONSE

I've encountered this issue both on normal and beta version of extension. Usually this happen after update is installed.
I haven't experienced such issue with any other extension in Vivaldi. In particular: Violentmonkey BETA, Stylus, Adblock Plus and uBlock Origin work fine after update and I never had an issue with opening their options.

Complete removal and reinstallation of extension fixes the issue, but all scripts have to be reinstalled and extension reconfigured (I personally change CSS, linter config and switch to instant injection mode). And it returns back after the next update.

This is likely Vivaldi bug since I never seen such problem in any other browser, but I have no idea what cause it and how to report it back to them.

## Specifications

- Vivaldi: 3.6.2165.36 (Stable channel) (64-bit)
- TM: BETA 4.12.6127 (same with stable version, just happens less often due to less often updates)
- OS: Windows 10 OS Version 2009 (Build 19042.804)

## Script

N/A

**Comment by derjanb** [comment]
I had a very old version installed which now updated to `3.7.2202.3` and it's working fine there. Do you still see this issue at 3.7?

![image](https://user-images.githubusercontent.com/767504/108610253-967d7f00-73d4-11eb-88c5-884a392741d1.png)

**Comment by lainverse** [comment]
Updated to latest 3.7.2202.3.
Dashboard failed to open.
Reinstalled Tampermonkey BETA.
Dashboard is operational again.
Will update when next version will be released (unless you know faster way to to check this).

**Comment by lainverse** [comment]
Upd, apparently it's unrelated to version update. Today I started Vivaldi again and this time Dashboard failed to open.
```
This site can’t be reachedThe webpage at chrome-extension://gcalenpjmijncebpfijmoaglllgpjagf/options.html#url=&nav=dashboard might be temporarily down or it may have moved permanently to a new web address.
ERR_INVALID_RESPONSE
```

**Comment by lainverse** [comment]
After a couple of edits in the script and restart of the browser dashboard broke again for no particular reason.
Extenstion's console output on page open:
```
background.js:192 Storage: use chromeStorage
background.js:182 Storage: test -> start
background.js:182 Storage: test -> set after 511ms
background.js:182 Storage: test -> get after 513ms
background.js:183 Storage: test -> remove after 1025ms
DevTools failed to load SourceMap: Could not load content for http://localhost:8000/sourcemaps/background.js.map: Connection error: net::ERR_CONNECTION_REFUSED
cache.js:5 SW: unexpected error Failed to fetch
```
crash is on `fetch(t.request);` here:
```
 if (!n.startsWith("http") || !(n.startsWith("https://www.google.com/s2/favicons") || n.startsWith("https://icons.duckduckgo.com/ip2/") || n.endsWith("favicon.ico")))
                    return fetch(t.request);
```
`m = "chrome-extension://gcalenpjmijncebpfijmoaglllgpjagf/options.html#url=&nav=dashboard"`
`r = undefined`

**Comment by lainverse** [comment]
And this is how it looks like when it works:
```
background.js:182 Storage: test -> start
background.js:182 Storage: test -> set after 508ms
background.js:182 Storage: test -> get after 509ms
background.js:183 Storage: test -> remove after 1022ms
Unchecked runtime.lastError: The message port closed before a response was received.
Unchecked runtime.lastError: The message port closed before a response was received.
Unchecked runtime.lastError: The message port closed before a response was received.
Unchecked runtime.lastError: The message port closed before a response was received.
Unchecked runtime.lastError: The message port closed before a response was received.
background.html:1 Unchecked runtime.lastError: The message port closed before a response was received.
background.html:1 Unchecked runtime.lastError: The message port closed before a response was received.
background.html:1 Unchecked runtime.lastError: The message port closed before a response was received.
background.html:1 Unchecked runtime.lastError: The message port closed before a response was received.
background.html:1 Unchecked runtime.lastError: The message port closed before a response was received.
DevTools failed to load SourceMap: Could not load content for http://localhost:8000/sourcemaps/background.js.map: Connection error: net::ERR_CONNECTION_REFUSED
An unknown error occurred when fetching the script.
An unknown error occurred when fetching the script.
```
Unknown error is generated by attempt to load chrome-extension://gcalenpjmijncebpfijmoaglllgpjagf/cache.js
So, basically, it works fine when cache.js doesn't.

**Comment by lainverse** [comment]
Updated to 4.12.6128. Dashboard opened at start, but then ceased to work in a minute.
Is there any logs I can provide to help find out what's wrong?

**Comment by lainverse** [comment]
Updated to 4.12.6129.
Dashboard link attempts to open this URL instead:
extension://gcalenpjmijncebpfijmoaglllgpjagf/options.html#url=aHR0cHM6Ly93d3cudGFtcGVybW9ua2V5Lm5ldC9jaGFuZ2Vsb2cucGhwP3ZlcnNpb249NC4xMi42MTI5JmV4dD1nY2FsJnVwZGF0ZWQ9dHJ1ZSZvbGQ9NC4xMi42MTI4JmludHI9dHJ1ZQ==&nav=dashboard
Which is as I understand a redirect to https://www.tampermonkey.net/changelog.php?version=4.12.6129&ext=gcal&updated=true&old=4.12.6128&intr=true

For some reason it doesn't even open the right link (even after browser restart). url parameter became empty only after I attempted to edit script directly. It failed to open a script for edit, but dashboard link got fixed... but still fails to open.

**Comment by emoryy** [comment]
I experience the same thing. The dashboard and other extension pages load fine for a few minutes. Then maybe after around 10 minutes of having (for example) an editor open, extension pages stop loading or reloading. For example: `chrome-extension://gcalenpjmijncebpfijmoaglllgpjagf/options.html#url=&nav=dashboard`
Scripts themselves continue to load and work though.

It's only fixed again temporarily by restarting the browser.

Tampermonkey: 4.12.6129
Vivaldi: 3.7.2218.3

**Comment by raszpl** [comment]
delete service worker (inspect/application/service workers)

this is related to #1079

**Comment by derjanb** [comment]
As a workaround: executing ```localStorage.setItem('no_sw', 'true');``` at the options page's console will disable the caching service worker.

**Comment by lainverse** [comment]
Workaround helps. Removal of the property with `localStorage.removeItem('no_sw');` returns worker back, but it fails to load worker script as I previously described (and actually registers 2 workers and then keeps registering new ones per successful dashboard open / refresh). Guess it will break as soon as that worker start to "work" again, so I'll keep using workaround for now. Thanks.

**Comment by raszpl** [comment]
Same with manually deleting Service Worker, now I get this in console:

    SQ: registration for /options.html failed -> TypeError: Failed to register a ServiceWorker for scope ('chrome-extension://gcalenpjmijncebpfijmoaglllgpjagf/options.html') with script ('chrome-extension://gcalenpjmijncebpfijmoaglllgpjagf/cache.js'): An unknown error occurred when fetching the script.

at least now it loads Dashboard fine :), but no service worker registered. Also Vivaldi, also 4.12.6132, probably happened when updating to 6132. It looks like Vivaldi is the culprit.

**Comment by vernilla** [comment]
I've just started experiencing this today, both before and after updating Vivaldi. Restarting makes it work for about five minutes, and then it fails again.

**Comment by lainverse** [comment]
@vernilla Try following workaround for now:

1. `Ctrl`+`Shift`+`E` (Open installed Extensions tab)
2. Enable `Developer Mode` in the top-right corner.
3. Click on `background.html` link of Tampermonkey extension.
4. Switch to `Console` tab in the newly opened window.
5. Copy this command `localStorage.setItem('no_sw', 'true');`, paste it there and press `Enter` to run it.

When issue eventually get fixed you can remove workaround with command `localStorage.removeItem('no_sw');` in the same window.

**Comment by rabe85** [comment]
Workaround helps, thanks a lot :)

**Comment by Trung0246** [comment]
> 5. Copy this command `localStorage.setItem('no_sw', 'true');`, paste it there and press `Enter` to run it.

I wonder what is "no_sw" btw?

**Comment by lainverse** [comment]
> I wonder what is "no_sw" btw?
No Service Worker. TM uses [service worker](https://developers.google.com/web/fundamentals/primers/service-workers/) for caching purposes, but it doesn't work right in Vivaldi.

**Comment by KarlPiper** [comment]
Just a note, when you get "This site can’t be reached" message, a hard refresh does force the page to load. Normal refresh does nothing.

**Comment by yattyatt** [comment]
I've been having this issue too on 4.13 and beta 4.14.6142 and vivaldi 3.8.2259.42 (Stable channel) (64-bit).  I've been unregistering the service worker manually to temporarily fix it.  Good to know there's a way to disable the service worker permanently.

**Comment by H4M5TER** [comment]
I had the same issue using Vivaldi today and got workaround here, thanks.
What can we do to solve the problem? Any information needed?
Shall we make a bug report to Vivaldi since there are no problems on other browsers?

**Comment by begna112** [comment]
> @H4M5TER I am a tester for Vivaldi and I've filed a bug report to them. Hopefully it will be resolved soon.

can you link it?

**Comment by raszpl** [comment]
> > @H4M5TER I am a tester for Vivaldi and I've filed a bug report to them. Hopefully it will be resolved soon.
> 
> can you link it?

Vivaldi doesnt have a public bug reporting system. The official way is you receiving a number and occasionally asking someone on their forum to check the sttus of the bug - this way they can hide number of bugs and the fact no one is working on them.

**Comment by Metalism666** [comment]
> @vernilla Try following workaround for now:
> 
> 1. `Ctrl`+`Shift`+`E` (Open installed Extensions tab)
> 2. Enable `Developer Mode` in the top-right corner.
> 3. Click on `background.html` link of Tampermonkey extension.
> 4. Switch to `Console` tab in the newly opened window.
> 5. Copy this command `localStorage.setItem('no_sw', 'true');`, paste it there and press `Enter` to run it.
> 
> When issue eventually get fixed you can remove workaround with command `localStorage.removeItem('no_sw');` in the same window.

After following your steps,the Console just says "undefined",the dashborad still cant be opened after restarting the browser : ( 
is there any other solution?

**Comment by m-thomson** [comment]
I'm seeing this bug as well. Super frustrating.

**Comment by brasofilo** [comment]
I'm having the same error running on macOS 10.14.6, TM 4.13 and Vivaldi 4.2.2406.44

I tried the workaround and it seemed not to work...  
Then I clicked "Create a new script" instead of "Dashboard" and the dashboard opened, but I'm not sure if this is what made it work again.

**Comment by liam-edwards** [comment]
Unsure if my issue is related, I'm getting `Failed to load resource: net::ERR_FILE_NOT_FOUND` as an error. On background.html and options.html pages. Settings `no_sw` to `true` doesn't resolve this either.

This has been happening since Friday when I restarted my PC, which I'm guessing updated the extension. Tried updating the browser today to no avail.

Running Tampermonkey BETA (4.14.6144) and Vivaldi (4.2.2406.52).

I would try to reinstall the extension but I'm worried I'll lose all my scripts if I do.
